src/FluidXPlate/FluidXScanner.php

```diff
@@ -0,0 +1,106 @@
+<?php declare(strict_types=1);
+
+namespace Mll\LiquidHandlingRobotics\FluidXPlate;
+
+use Illuminate\Support\Str;
+use Mll\Microplate\Coordinate;
+use Mll\Microplate\CoordinateSystem96Well;
+use MLL\Utils\StringUtil;
+
+/**
+ * Communicates with a FluidX scanner device and fetches results from it.
+ */
+class FluidXScanner
+{
+    private const READING = 'Reading...';
+    private const XTR_96_CONNECTED = 'xtr-96 Connected';
+    private const NO_READ = 'NO READ';
+    private const NO_TUBE = 'NO TUBE';
+    public const LOCALHOST = '127.0.0.1';
+
+    public function scanPlate(string $ip): FluidXPlate
+    {
+        if (self::LOCALHOST === $ip) {
+            return self::returnTestPlate();
+        }
+
+        if ('' === $ip) {
+            throw new ScanFluidXPlateException('Cannot start scan request without an IP address.');
+        }
+
+        try {
+            $socket = \Safe\fsockopen($ip, 8001, $errno, $errstr, 30);
+        } catch (\Throwable $e) {
+            throw new ScanFluidXPlateException("Cannot reach FluidX Scanner {$ip}: {$e->getMessage()}. Verify that the FluidX Scanner is turned on and the FluidX software is started.", 0, $e);
+        }
+
+        \Safe\fwrite($socket, "get\r\n");
+
+        $answer = '';
+        do {
+            $content = fgets($socket);
+            $answer .= $content;
+        } while (is_string($content) && ! Str::contains($content, 'H12'));
+
+        \Safe\fclose($socket);
+
+        return self::parseRawContent($answer);
+    }
+
+    public static function parseRawContent(string $rawContent): FluidXPlate
+    {
+        if ('' === $rawContent) {
+            throw new ScanFluidXPlateException('Der Scanner lieferte ein leeres Ergebnis zurück.');
+        }
+
+        $lines = StringUtil::splitLines($rawContent);
+        $barcodes = [];
+        $id = null;
+        foreach ($lines as $line) {
+            if ('' !== $line
+                && self::READING !== $line
+                && self::XTR_96_CONNECTED !== $line
+            ) {
+                $content = explode(', ', $line);
+
+                if (count($content) > 3) {
```

Comment:
I guess my comment was not clear. What I meant is that the flying V in this section can be reduced by using early returns (in this case `continue`).

Reply:
done